backend/routes/admin_routes.py

```
@@ -122,13 +129,21 @@ def admin_login():
         }), 200
 
     except Exception as e:
-        logging.error("Errore durante il login: %s", e)
+        logger.error("Login error occurred", extra=safe_extra_fields({
+            'username': data.get('username'),
+            'error_type': type(e).__name__,
+            'error_details': str(e),
+            'operation_result': 'failed'
+        }))
```

Ruff (0.13.1)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)